src/key.js

```diff
@@ -771,6 +771,118 @@ Key.prototype.verifyAllUsers = async function(keys) {
   return results;
 };
 
+/**
+ * generate a new subkey to a Key
+ * @param  {Object} options   (optional) options for creating subkey, e.g. {sign: true, passphrase: '123', algorithm:..., curve:...}
+ *
+ * @returns {Promise<module:key.SubKey>} return the subkey if successful.
+ * @async
+ */
+Key.prototype.generateSubkey = async function(options){
+  options = sanitizeKeyOption(options, this.primaryKey.algorithm);
+  const secretSubKeyPacket = await generateSecretSubkey(options);
+  const result = await this.addSubkey(secretSubKeyPacket, options);
+  return result;
+
+  function sanitizeKeyOption(options, algo) {
```

Comment:
This is basically a duplicate of https://github.com/openpgpjs/openpgpjs/blob/master/src/key.js#L1228. Rather than duplicating this code, I suggest that you pull out the `sanitizeKeyOptions` function from key generation and use it here as well, with any necessary modifications.

Reply:
The `sanitizeKeyOptions` of  the subkey generation should try to guess the curve or rsa from the primaryKey. But the main key generation would not.

Reply:
Understood. But this means that those sanitization functions should probably be pulled out of their parent functions and decomposed so that you can use parts of them in both cases.